components/leave_team_modal/index.ts

```diff
@@ -4,36 +4,50 @@
 import {connect} from 'react-redux';
 import {bindActionCreators, Dispatch} from 'redux';
 
-import {getCurrentUserId, getCurrentUser} from 'mattermost-redux/selectors/entities/users';
-import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
-import {getPrivateChannels, getPublicChannels} from 'mattermost-redux/selectors/entities/channels';
 import {removeUserFromTeam as leaveTeam} from 'mattermost-redux/actions/teams';
+
+import {getMyChannels} from 'mattermost-redux/selectors/entities/channels';
+import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
+import {getCurrentUserId, getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+
 import {GenericAction} from 'mattermost-redux/types/actions';
 
 import {toggleSideBarRightMenuAction} from 'actions/global_actions';
-import {ModalIdentifiers} from 'utils/constants';
 
 import {isModalOpen} from 'selectors/views/modals';
 
 import {GlobalState} from 'types/store';
 
+import {Constants, ModalIdentifiers} from 'utils/constants';
+
 import LeaveTeamModal from './leave_team_modal';
 
+function getNumOfPrivateChannels(state: GlobalState) {
+    const channels = getMyChannels(state);
+
+    return channels.filter((channel) => channel.type === Constants.PRIVATE_CHANNEL).length;
+}
+
+function getNumOfPublicChannels(state: GlobalState) {
+    const channels = getMyChannels(state);
+
+    return channels.filter((channel) => channel.type === Constants.OPEN_CHANNEL).length;
+}
+
 function mapStateToProps(state: GlobalState) {
     const modalId = ModalIdentifiers.LEAVE_TEAM;
     const currentUserId = getCurrentUserId(state);
     const currentTeamId = getCurrentTeamId(state);
-    const privateChannels = getPrivateChannels(state);
-    const publicChannels = getPublicChannels(state);
     const show = isModalOpen(state, modalId);
     const currentUser = getCurrentUser(state);
+
     return {
         currentUserId,
         currentTeamId,
         show,
         currentUser,
-        privateChannels,
-        publicChannels,
+        numOfPrivateChannels: getNumOfPrivateChannels(state),
```

Comment:
Switching these to pass plain numbers wasn't strictly necessary since the main thing was that I was trying to get rid of the selectors that were used here before.